File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer =new int[2];
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;
        int a = numer;
        int b = denom;
        
        while(b != 0){
            int temp = b;
            b = a % b;
            a = temp;
        }
        
        answer[0] = numer / a;
        answer[1] = denom / a;
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math 

def solution(numer1, denom1, numer2, denom2):
    answer = []
    deno = denom1 * denom2 # 분모
    numer = denom1 * numer2 + denom2 * numer1
    
    gcd = math.gcd(deno, numer)
    
    answer.append(numer/gcd)
    answer.append(deno/gcd)
    return answer